langchain4j-azure-cosmos-no-sql/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/no/sql/AzureCosmosDBNoSqlEmbeddingStore.java

```diff
@@ -0,0 +1,215 @@
+        this.containerProperties = containerProperties;
+
+        if (cosmosClient == null) {
+            throw new IllegalArgumentException("cosmosClient cannot be null or empty for Azure CosmosDB NoSql Embedding Store.");
+        }
+
+        if (isNullOrBlank(databaseName) || isNullOrBlank(containerName)) {
+            throw new IllegalArgumentException("databaseName and containerName needs to be provided.");
+        }
+
+        if (cosmosVectorEmbeddingPolicy == null || cosmosVectorEmbeddingPolicy.getVectorEmbeddings() == null ||
+                cosmosVectorEmbeddingPolicy.getVectorEmbeddings().isEmpty()) {
+            throw new IllegalArgumentException("cosmosVectorEmbeddingPolicy cannot be null or empty for Azure CosmosDB NoSql Embedding Store.");
+        }
+
+        if (cosmosVectorIndexes == null || cosmosVectorIndexes.isEmpty()) {
+            throw new IllegalArgumentException("cosmosVectorIndexes cannot be null or empty for Azure CosmosDB NoSql Embedding Store.");
+        }
+
+
+        cosmosClient.createDatabaseIfNotExists(this.databaseName);
+        this.database = this.cosmosClient.getDatabase(this.databaseName);
+
+        containerProperties.setVectorEmbeddingPolicy(this.cosmosVectorEmbeddingPolicy);
+        containerProperties.getIndexingPolicy().setVectorIndexes(this.cosmosVectorIndexes);
+
+        this.database.createContainerIfNotExists(this.containerProperties);
+        this.container = this.database.getContainer(this.containerName);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public String add(Embedding embedding) {
+        String id = randomUUID();
+        add(id, embedding);
+        return id;
+    }
+
+    @Override
+    public void add(String id, Embedding embedding) {
+        addInternal(id, embedding, null);
+    }
+
+    @Override
+    public String add(Embedding embedding, TextSegment textSegment) {
+        String id = randomUUID();
+        addInternal(id, embedding, textSegment);
+        return id;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(Collectors.toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(Collectors.toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+
+        String embeddingKey = this.cosmosVectorEmbeddingPolicy.getVectorEmbeddings().get(0).getPath().substring(1);
```

Comment:
What is embedding key? This can probably be done only once, in the constructor?

Reply:
Fixed in new iteration.